docs/AUTH_PROTOCOL.md

```diff
@@ -0,0 +1,393 @@
+    - The frontend should stop all of its mechanisms (including secondary mechanisms) whenever it requests the start of a new flow
+    - Arguments:
+        - `flow`: The flow to start. It's perfectly valid for this to be the same as the flow that's currently active
+    - Response: The authenticator starts the conversation and sends a primary authentication mechanism
+
+- `flowChanged`: Notifies the frontend that the authenticator has changed the user's flow
+    - Send from the authenticator to the frontend
+    - The frontend should act as if the user has selected a new flow from the menu, except it should NOT emit a `start` event and should NOT interrupt any mechanisms
+    - The frontend should update the user's remembered preferred flow
+    - Arguments:
+        - `flow`: The flow to change to
+    - Response: None
+
+#### Password change events
+
+These events might be emitted while the `newPassword` mechanism is active:
+
+- `newPasswordRejected`: The new password has failed for some reason. Try again
+    - Sent from authenticator to frontend
+    - The frontend should notify the user that the new password has been rejected and that they should pick a different password. The `newPassword` mechanism should remain active
+    - Arguments:
+        - `reason`: The reason that the new password was rejected. Valid reasons follow:
+            - `custom`: Just use the `fallbackMessage`
+        - `fallbackMessage`: Fallback string to show in the UI if it doesn't support the provided reason
+    - Response: none. The `newPassword` mechanism will eventually emit a `response` as usual.
+
+#### Fingerprint events
+
+These events might be emitted while the `fingerprint` mechanism is active:
+
+- `fingerprintFeedback`: The fingerprint scanner is operational and has some feedback to report to the user
+    - Sent from authenticator to frontend
+    - The frontend should react by indicating to the user what is happening or what the user should do (i.e. by playing an animation, or giving written instructions)
+    - Arguments:
+        - `feedback`: The feedback to provide to the user. Valid values follow:
+            - `scanning`: The fingerprint sensor has detected a finger and is working to scan it
+            - `swipeTooSlow`: The user swiped across the sensor too slowly and needs to speed up
+            - `swipeTooFast`: The user swiped across the sensor too quickly and needs to slow down
+            - `morePressure`: The user isn't applying enough pressure on the fingerprint sensor
+            - `moveLeft`: The user should move their finger a little to the left to center it on the sensor
+            - `moveRight`, `moveUp`, `moveDown`: Like `moveLeft`, but for the other directions
+            - `recenter`: The user's finger is not centered on the sensor, but we don't have specific directional instructions available
+            - `cancelled`: The user has removed their finger from the fingerprint sensor before the scan could complete
```

Comment:
It's more "not enough" data, that we have in fprintd as "too-short" or "too-fast"